modal/_utils/blob_utils.py

```diff
@@ -165,6 +177,7 @@ async def perform_multipart_upload(
     part_urls: List[str],
     completion_url: str,
     upload_chunk_size: int = DEFAULT_SEGMENT_CHUNK_SIZE,
+    progress_report_cb: Callable,
```

Comment:
```suggestion
    progress_report_cb: Optional[Callable],
```

because it's called by `_blob_upload` which itself has an optional param. 

I think this might not be caught because this function is missing its return type annotation, which should be `-> None`